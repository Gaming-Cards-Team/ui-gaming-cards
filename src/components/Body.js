import React from "react";
import Banner from "./Banner";
import TopMessage from "./TopMessage";
import CardGallery from "./CardGallery";

const Body = () => {
  return (
    <div>
      <Banner></Banner>
      <TopMessage></TopMessage>
     
      <CardGallery></CardGallery>
    </div>
  );
};

export default Body;
