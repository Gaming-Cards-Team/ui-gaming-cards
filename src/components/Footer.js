import React from "react";
import Navbar from "react-bootstrap/Navbar";
import "bootstrap/dist/css/bootstrap.min.css";
import { FOOTER_TEXT } from "../util/constant.js"

const textStyle={
    color: "white",
    width: "100%",
    textAlign: "center",
  };
  const footerStyle={
    backgroundColor: "#2d963f"
  };

const Footer = () => (
  <Navbar 
  fixed="bottom"
  style={footerStyle}
  >
    <div
      data-testid="footer-text"
      style={textStyle}
    >
      {FOOTER_TEXT}
    </div>
  </Navbar>
);

export default Footer;
