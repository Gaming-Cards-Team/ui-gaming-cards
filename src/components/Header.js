import React from "react";
import logo from "../assets/images/logo.png";
import Navbar from "react-bootstrap/Navbar";
import "bootstrap/dist/css/bootstrap.min.css";
import { LOGO_ALTERNATIVE_TEXT } from "../util/constant";

const Header = () => {
  return (
    <Navbar bg="light" variant="light">
      <Navbar.Brand href="#home">
        <img
          data-testid="logo"
          height="48"
          src={logo}
          alt={LOGO_ALTERNATIVE_TEXT}
        />
      </Navbar.Brand>
    </Navbar>
  );
};

export default Header;
