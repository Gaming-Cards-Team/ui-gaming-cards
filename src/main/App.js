import React, { useState, useEffect } from 'react';
import MessageList from './MessageList';
import {PRODUCT_GALLERY_MANAGER} from './util/constant'

const App = () => {

  const [gallery, setGallery] = useState([]);

  useEffect(() => {
      
      fetch(PRODUCT_GALLERY_MANAGER)
      .then(res => res.json())
      .then(readResponse)
      .catch(console.log)
  });

  function readResponse(cards) {
    const gallery = cards.map(mapToItem)
    setGallery(gallery);
  }

  function mapToItem(card) {
    const item = { key: card._id, value: card.name } 
    return item;
  }

  return (
    <div>
      Cards:
      <MessageList data={gallery} />
    </div>
  );
};

export default App;