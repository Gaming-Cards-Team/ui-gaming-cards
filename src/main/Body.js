import React from 'react';

const Body = () => (
    <div id='body'>
       Body
    </div>
);

export default Body;