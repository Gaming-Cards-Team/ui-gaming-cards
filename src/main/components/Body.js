import React from "react";
import Banner from "./Banner";

const Body = () => {
  return (
      <Banner></Banner>
  );
};

export default Body;
