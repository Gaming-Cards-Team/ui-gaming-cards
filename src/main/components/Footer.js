import React from "react";
import Navbar from "react-bootstrap/Navbar";
import "bootstrap/dist/css/bootstrap.min.css";

const Footer = () => (
  <Navbar bg="success" fixed="bottom">
    <div
      data-testid="footer-text"
      style={{
        color: "white",
        width: "100%",
        textAlign: "center",
      }}
    >
      Star Wars 2020 All rights reserved. Copyright Level Upâ€©
    </div>
  </Navbar>
);

export default Footer;
