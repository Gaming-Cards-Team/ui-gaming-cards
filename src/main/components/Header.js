import AppBar from "@material-ui/core/AppBar";
import { makeStyles } from "@material-ui/core/styles";
import Toolbar from "@material-ui/core/Toolbar";
import React from "react";
import logo from "../resources/logo.png"

const useStyles = makeStyles({
  root: {
    flexGrow: 1
  },
  logo: {
    width: 135,
    height: 43.54
  }
});

const Header = () => {
  const classes = useStyles();
  return (
    <nav className={classes.root} id='header'>
      <AppBar position="static" color="default">
        <Toolbar>
          <img
            data-testid='logo'
            className={classes.logo}
            src={logo}
            alt="Level Up Logo"
          />
        </Toolbar>
      </AppBar>
    </nav>
  );
};

export default Header;
