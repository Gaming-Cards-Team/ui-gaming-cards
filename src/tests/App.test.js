import React from 'react';
import App from '../main/App';
import { mount, shallow } from 'enzyme';
import Header from '../main/components/Header';
import Body from '../main/components/Body';
import Footer from '../main/components/Footer';

describe('Test App component', () => {

    it('should have Header, Body, Footer', () => {

        const app  = mount(<App />)
        expect(app.containsAllMatchingElements([<Header/>, <Body/>, <Footer/>])).toBe(true)
    });

    it('should validate the component', () => {

        const app  = shallow(<App />)
        expect(app).toMatchSnapshot()
    });
});