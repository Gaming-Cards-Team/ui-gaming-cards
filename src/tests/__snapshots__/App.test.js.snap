// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component should validate the component 1`] = `
<div>
  <nav
    className="navbar navbar-expand navbar-light bg-light fixed-top"
  >
    <a
      className="navbar-brand"
      href="#home"
    >
      <img
        alt="Level Up Logo"
        className=""
        data-testid="logo"
        height="48"
        src="logo.png"
      />
    </a>
  </nav>
  <img
    alt="Star Wars Banner"
    className=""
    data-testid="banner"
    src="banner.jpg"
    style={
      Object {
        "opacity": 0.6,
      }
    }
    width="100%"
  />
  <nav
    className="navbar navbar-expand navbar-light bg-success fixed-bottom"
  >
    <div
      data-testid="footer-text"
      style={
        Object {
          "color": "white",
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      Star Wars 2020 All rights reserved. Copyright Level Upâ€©
    </div>
  </nav>
</div>
`;
